src/SLCore.Listeners/Implementation/AnalysisListener.cs

```diff
@@ -29,18 +29,42 @@ namespace SonarLint.VisualStudio.SLCore.Listeners.Implementation;
 [PartCreationPolicy(CreationPolicy.Shared)]
 internal class AnalysisListener : IAnalysisListener
 {
+    private readonly IAnalysisService analysisService;
+    private readonly IRaiseIssueParamsToAnalysisIssueConverter raiseIssueParamsToAnalysisIssueConverter;
+
+    [ImportingConstructor]
+    public AnalysisListener(IAnalysisService analysisService, IRaiseIssueParamsToAnalysisIssueConverter raiseIssueParamsToAnalysisIssueConverter)
+    {
+        this.analysisService = analysisService;
+        this.raiseIssueParamsToAnalysisIssueConverter = raiseIssueParamsToAnalysisIssueConverter;
+    }
+
     public Task DidChangeAnalysisReadinessAsync(DidChangeAnalysisReadinessParams parameters)
     {
         return Task.CompletedTask;
     }
 
     public void RaiseIssues(RaiseIssuesParams parameters)
     {
-        throw new NotImplementedException();
+        if (parameters.analysisId.HasValue)
+        {
+            var filepath = parameters.issuesByFileUri.Single().Key.LocalPath;
```

Comment:
This will throw for empty lists. IMO it's better to explicitly check that there's not 0, not 2, but only 1 file path and handle other cases with logging

Reply:
We should not have any scenarios in which this would happen. We would trigger analysis for a single file, and it would always have one item. I am not sure defensive approach is the best option here.